tests/drivers/clock_control/pwm_clock/src/main.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2023 Andriy Gelman <andriy.gelman@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/ztest.h>
+
+#define NODELABEL DT_NODELABEL(samplenode)
+static const struct device *clk_dev = DEVICE_DT_GET(DT_CLOCKS_CTLR(NODELABEL));
+
+static void *pwm_clock_setup(void)
+{
+	int ret;
+
+	zassert_equal(device_is_ready(clk_dev), true, "%s: PWM clock device is not ready",
+		      clk_dev->name);
+
+	ret = clock_control_on(clk_dev, 0);
+	zassert_equal(0, ret, "%s: Unexpected err (%d) from clock_control_on", clk_dev->name, ret);
+
+	return NULL;
+}
+
+ZTEST(pwm_clock, test_clock_control_get_rate)
+{
+	int ret;
+	uint32_t clock_rate;
+
+	ret = clock_control_get_rate(clk_dev, 0, &clock_rate);
+	zassert_equal(0, ret, "%s: Unexpected err (%d) from clock_control_get_rate",
+		      clk_dev->name, ret);
```

Comment:
Since you cannot validate the returned rate here (as `test_clock_control_set_rate` may have been calledf first), I suggest you add a `setup` method validating the initial clock rate being set via devicetree.